chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts

```diff
@@ -1,104 +1,98 @@
+        dispatch({ type: LiveChatWidgetActionType.SET_RECONNECT_ID, payload: reconnectChatContext.reconnectId ?? "" });
+        dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.ReconnectChat });
+        return;
     }
 };
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const startUnauthenticatedReconnectChat = async (chatSDK: any, chatConfig: ChatConfig | undefined, getAuthToken: ((authClientFunction?: string) => Promise<string | null>) | undefined, dispatch: Dispatch<ILiveChatWidgetAction>, setAdapter: any, reconnectId: string, initStartChat: any) => {
-    const reconnectAvailabilityResponse: IReconnectChatContext = await getChatReconnectContext(chatSDK, chatConfig, getAuthToken, reconnectId);
-    if (!shouldRedirectOrStartNewChat(reconnectAvailabilityResponse)) {
-        await setReconnectIdAndStartChat(chatSDK, chatConfig, getAuthToken, dispatch, setAdapter, reconnectId, initStartChat);
+const getChatReconnectContext = async (chatSDK: any, chatConfig: ChatConfig, props: any, isAuthenticatedChat: boolean) => {
+    try {
+        const chatReconnectOptionalParams: IReconnectChatOptionalParams = {
+            reconnectId: props.reconnectChatPaneProps?.reconnectId
+        };
+        if (isAuthenticatedChat) {
+            // Get auth token for for getting chat reconnect context
+            await handleAuthentication(chatSDK, chatConfig, props.getAuthToken);
+        }
+        const reconnectChatContext = await chatSDK?.getChatReconnectContext(chatReconnectOptionalParams);
+        if (isAuthenticatedChat) {
+            // remove auth token after reconnectId is fetched
+            // this will be reset later at start chat
+            removeAuthTokenProvider(chatSDK);
+        }
+        return reconnectChatContext;
+    }
+    catch (error) {
+        TelemetryHelper.logSDKEvent(LogLevel.ERROR, {
+            Event: TelemetryEvent.GetChatReconnectContextSDKCallFailed,
+            ExceptionDetails: {
+                exception: error
+            }
+        });
     }
 };
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const setReconnectIdAndStartChat = async (chatSDK: any, chatConfig: ChatConfig | undefined, getAuthToken: ((authClientFunction?: string) => Promise<string | null>) | undefined, dispatch: Dispatch<ILiveChatWidgetAction>, setAdapter: any, reconnectId: string, initStartChat: any) => {
-    const startUnauthenticatedReconnectChat: ICustomEvent = {
-        eventName: BroadcastEvent.StartUnauthenticatedReconnectChat,
-    };
-    BroadcastService.postMessage(startUnauthenticatedReconnectChat);
+const setReconnectIdAndStartChat = async (isAuthenticatedChat: boolean, chatSDK: any, chatConfig: ChatConfig | undefined, getAuthToken: ((authClientFunction?: string) => Promise<string | null>) | undefined, dispatch: Dispatch<ILiveChatWidgetAction>, setAdapter: any, reconnectId: string, initStartChat: any) => {
```

Comment:
curious about the mixup of optional parameters and required parameters using as default undefined value, 

is there a way to reduce verbosity and make it more clean?,

I'd suggest to explore the use of objects wrapper for multiple input parameters,

Reply:
I agree. I tried fixing few places but there are still other occurrences which needs to be fixed by original author of this code to avoid any regression. We can create a task to review these.